pkgs/by-name/bc/bcachefs-tools/package.nix

```diff
@@ -97,6 +99,11 @@ stdenv.mkDerivation (finalAttrs: {
   postInstall = ''
     substituteInPlace $out/libexec/bcachefsck_all \
       --replace-fail "/usr/bin/python3" "${python3}/bin/python3"
+
```

Comment:
Could you guard the new lines with `lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform)`? This will prevent build errors in cross compilation.

Reply:
Now the substituteInPlace is also not executed for cross builds, I was thinking more along the lines of 

```
postInstall = ''
    substituteInPlace $out/libexec/bcachefsck_all \
      --replace-fail "/usr/bin/python3" "${python3}/bin/python3"
'' + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
    installShellCompletion --cmd bcachefs \
      --bash <($out/sbin/bcachefs completions bash) \
      --zsh  <($out/sbin/bcachefs completions zsh) \
      --fish <($out/sbin/bcachefs completions fish)
'';
```

Reply:
Now the substituteInPlace is also not executed for cross builds, I was thinking more along the lines of 

```
postInstall = ''
    substituteInPlace $out/libexec/bcachefsck_all \
      --replace-fail "/usr/bin/python3" "${python3}/bin/python3"
'' + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
    installShellCompletion --cmd bcachefs \
      --bash <($out/sbin/bcachefs completions bash) \
      --zsh  <($out/sbin/bcachefs completions zsh) \
      --fish <($out/sbin/bcachefs completions fish)
'';
```

I've tested the build with `nix build .#bcachefs-tools` and `nix build .#pkgsCross.aarch64-multiplatform.bcachefs-tools` (from `x86_64-linux`).